### PR TITLE
ci: init docker images publication pipeline generation [skip ci]

### PR DESCRIPTION
**Pull request by Okhelifi**
## Issue

https://gravitee.atlassian.net/browse/APIM-2635

## Description

Docker image pipeline build using circle ci sdk

### CODE REVIEW

File: .circleci/ci/src/jobs/job-setup.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { config } from '../config';
+import { CircleCIEnvironment } from '../pipelines';
+import { RestoreMavenCache } from '../commands';
+
+export class SetupJob {
+  public static create(dynamicConfig: Config, environment: CircleCIEnvironment): Job {
+    const openJdkExecutor = OpenJdkExecutor.get('small');
+    dynamicConfig.addReusableExecutor(openJdkExecutor);
+
+    const restoreMavenCacheCmd = RestoreMavenCache.get('job-build', environment);
```
Comment (phiz71) [inline .circleci/ci/src/jobs/job-setup.ts]: useless

File: .circleci/ci/src/jobs/job-setup.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      }),
+      new commands.Run({
+        command: `echo $MAVEN_SETTINGS > ${config.maven.settingsFile} `,
+      }),
+      new commands.workspace.Persist({
+        root: '.',
+        paths: [config.maven.settingsFile],
+      }),
+    ];
+    return new Job('job-build-backend', new ReusedExecutor(openJdkExecutor), steps);
```
Comment (phiz71) [inline .circleci/ci/src/jobs/job-setup.ts]: The job name is not the right one.
```suggestion
    return new Job('job-setup', new ReusedExecutor(openJdkExecutor), steps);
```

On top of that, we don't need a JDK executor, as we don't use maven to determine the version. So probably the simplest executor is enough.

File: .circleci/ci/src/utils/tests/resources/pom-sha1-snapshot.xml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    </scm>
+
+    <issueManagement>
+        <url>https://github.com/gravitee-io/issues/issues</url>
+        <system>GitHub Issues</system>
+    </issueManagement>
+    <properties>
+        <!-- Version properties -->
+        <?xml version="1.0" encoding="UTF-8"?>
+        <!--
```
Comment (phiz71) [inline .circleci/ci/src/utils/tests/resources/pom-sha1-snapshot.xml]: 🤔

File: .circleci/ci/src/utils/tests/resources/pom-sha1-snapshot.xml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+            <issueManagement>
+                <url>https://github.com/gravitee-io/issues/issues</url>
+                <system>GitHub Issues</system>
+            </issueManagement>
+            <properties>
+                <!-- Version properties -->
+                <revision>4.2.0</revision>
+                <sha1/>
+                <changelist/>
```
Comment (phiz71) [inline .circleci/ci/src/utils/tests/resources/pom-sha1-snapshot.xml]: Copy&Paste problem

File: .circleci/ci/src/utils/versions.ts [hunk 5]
```diff
@@ -32,3 +35,22 @@ export function parse(graviteeioVersion: string): GraviteeioVersion {
     },
   };
 }
+
+export async function computeApimVersion(pomXmlPath = '~/projects/gravitee-api-management/pom.xml'): Promise<string> {
+  // Read the pom.xml file
+  const pomXml = fs.readFileSync(pomXmlPath, 'utf8');
+
+  // Define the constants
+  let revision = '';
```
Comment (jourdiw) [inline .circleci/ci/src/utils/versions.ts]: I'm not sure if this part helps? We only use the values once in the return.

**Review by jourdiw — APPROVED** [review]

**Review by jourdiw — APPROVED** [review]
LGTM

**Review by phiz71 — APPROVED** [review]

**Pull request merged**